SourceStream/file/checksum.py

```diff
@@ -0,0 +1,37 @@
+from bs4 import BeautifulSoup as bs
+import requests
+
+
+def checksum(): 
+    # gets page contents
+    request = requests.get('https://linuxfromscratch.org/lfs/view/stable-systemd/chapter03/packages.html', headers={'User-Agent': 'Mozilla/5.0'})
+    if request.status_code != 200: 
+        print('Error: ' + str(request.status_code))      
+    page = bs(request.content, 'lxml')
+    list = page.find('dl')
+    
+    # gets package names
+    names = []
+    for item in list.find_all('dt'):
+        name = item.text.strip().split('(')[0]
+        names.append(name)
+    
+    versions = []
+    for item in list.find_all('dt'):
+        version = item.text.strip().split('(')[1].split(')')[0]
+        versions.append(version)
+    
+    #gets package checksums
+    checksums = []
+    for item in list.find_all('dd'):
+        checksum = item.find('code').text
+        checksums.append(checksum)
+    
+    # writes checksums to file
+    f = open('SourceStream/file/checksums.txt', 'w')
+    f.write(''.join(f'{names[i]}: {versions[i]}: {checksums[i]}\n' for i in range(len(names))))
+    f.close()
+    
+    
+if __name__ == '__main__':
```

Comment:
Instead of having basically 2 seperate programs maybe try to call this file from SourceStream.py